pandas/tests/indexes/common.py

```diff
@@ -682,6 +682,21 @@ def test_hasnans_isnans(self):
                 self.assert_numpy_array_equal(idx._isnan, expected)
                 self.assertTrue(idx.hasnans)
 
+    def test_ensure_copied_data(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, (PeriodIndex, RangeIndex, MultiIndex)):
+                # Cannot be instantiated from just _data as first argument
+                continue
+
+            data = index._data.copy()
+            idx_type = index.__class__
+            idx = idx_type(data, copy=True)
+            self.assert_numpy_array_equal(data, idx._data)
```

Comment:
I added this test is precisely for data (i.e. to test ``copy=True``, see the ``assertIsNot`` at the following line). If you mean this specific _line_ is not testing any of this, then I agree, it was just cheap to add, but I can remove it.